packages/gafl-webapp-service/src/handlers/__tests__/check-analytics-handler.spec.js

```diff
@@ -13,20 +16,44 @@ describe('checkAnalytics', () => {
     jest.clearAllMocks()
   })
 
+  it.each`
+    desc                                                                                           | analytics                                                           | page     | trackingResult
+    ${'of accept tracking as true, page should not be skipped then return checkAnalytics as true'} | ${{ [ANALYTICS.acceptTracking]: true }}                             | ${false} | ${true}
+    ${'of accept tracking as true, page should be skipped then return checkAnalytics as false'}    | ${{ [ANALYTICS.acceptTracking]: true, [ANALYTICS.skipPage]: true }} | ${true}  | ${false}
+    ${'but accept tracking is false so return checkAnalytics as false'}                            | ${{ [ANALYTICS.acceptTracking]: false }}                            | ${false} | ${false}
+  `('when analytics has a value $desc', async ({ analytics, page, trackingResult }) => {
+    const result = await checkAnalytics(generateRequestMock(analytics), page)
+
+    expect(result).toBe(trackingResult)
+  })
+
+  it('empty session cache returns false', async () => {
+    const result = await checkAnalytics(
+      generateRequestMock(() => {
+        throw new CacheError()
+      })
+    )
+    expect(result).toEqual(false)
+  })
+})
+
+describe('checkPage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
   it.each([
     [true, true],
-    [false, false],
-    [undefined, false]
-  ])('returns value of [ANALYTICS.acceptTracking]', async (acceptTracking, trackingResult) => {
-    const analytics = () => ({
-      [ANALYTICS.acceptTracking]: acceptTracking
-    })
-    const result = await checkAnalytics(generateRequestMock(analytics))
-    expect(result).toEqual(trackingResult)
+    [false, false]
+  ])('when analytics skipPage property is %s, skipPage returns %s', async (skipPage, expected) => {
+    const analytics = { [ANALYTICS.skipPage]: skipPage }
+    const result = await checkPage(generateRequestMock(analytics))
+
+    expect(result).toBe(expected)
   })
 
   it('empty session cache returns false', async () => {
-    const result = await checkAnalytics(
+    const result = await checkPage(
```

Comment:
Because of this change, you're no longer checking the try...catch in `checkAnalytics`